bleak/backends/corebluetooth/PeripheralDelegate.py

```diff
@@ -196,7 +197,7 @@ async def startNotify_cb_(
         event = self._characteristic_notify_change_events.get_cleared(c_handle)
         self.peripheral.setNotifyValue_forCharacteristic_(True, characteristic)
         # wait for peripheral_didUpdateNotificationStateForCharacteristic_error_ to set event
-        # await event.wait()
+        await event.wait()
```

Comment:
Why was this event waiting commented out? It feels like it must have been done due to some reason once...

Reply:
As mentioned in the commit message, it was commented out in f3bead7 and was unrelated to that change, which was about fixing RSSI.

Reply:
@Carglglz, you remember why this was done?

Reply:
Bleak objects cannot be used from threads other than the one where they were created (not needing to do this is one of the benefits of using asyncio). There is `call_soon_threadsafe()` to schedule something on an event loop from a different thread if needed.

Could you open a new issue for the hang when stopping notifications with a reproducible code example?

Reply:
> Also how does peripheral_didUpdateNotificationStateForCharacteristic_error_ to set event sets the event flag to True if event is not passed to self.peripheral.setNotifyValue_forCharacteristic_ ?

Ok, I remember now how it is done

```
def did_update_notification_for_characteristic(
        self, peripheral: CBPeripheral, characteristic: CBCharacteristic, error: NSError
    ):
        cUUID = characteristic.UUID().UUIDString()
        c_handle = characteristic.handle()
        if error is not None:
            raise BleakError(
                "Failed to update the notification status for characteristic {}: {}".format(
                    cUUID, error
                )
            )
        logger.debug("Character Notify Update")

        event = self._characteristic_notify_change_events.get(c_handle)
        if event:
            event.set()
        else:
            logger.warning(
                "Unexpected event didUpdateNotificationStateForCharacteristic"
            )

def peripheral_didUpdateNotificationStateForCharacteristic_error_(
        self, peripheral: CBPeripheral, characteristic: CBCharacteristic, error: NSError
    ):
        logger.debug("peripheral_didUpdateNotificationStateForCharacteristic_error_")
        self._event_loop.call_soon_threadsafe(
            self.did_update_notification_for_characteristic,
            peripheral,
            characteristic,
            error,
        )
```

I see now that ``self._event_loop.call_soon_threadsafe`` could  be the reason why it was blocking. I was handling notifications in another thread with a new event loop. I will test this changes , see If it is fixed or what to do to avoid commenting out ``await event.wait()`` and works with threads too.

Reply:
> Could you open a new issue for the hang when stopping notifications with a reproducible code example?
Yes, although the reproducible code example may be a little too complicated. 

See https://github.com/Carglglz/bleico, 

https://github.com/Carglglz/bleico/blob/1f0907a8425f2db4ffb10666b87ec7c7fc63ba22/bleico/systrayicon.py#L964

Here I'm trying to start and stop notifications in a GUI application (systray icon to be precise)... I may be a better way to do this but at that moment I didn't find any.